src/functional.jl

```diff
@@ -0,0 +1,111 @@
+module FunctionalNullableOperations
+
+# extends Base with operations that treat nullables as collections
+
+importall Base
+import Base: promote_op, LinearFast
+
+# conceptually, nullable types can be considered infinite-order tensor powers of
```

Comment:
I don't think this comment really helps understanding the code. Personally, I rather find it hard to get. :-)